Objects/typeobject.c

```diff
@@ -6957,9 +7004,18 @@ update_one_slot(PyTypeObject *type, slotdef *p)
         } while (p->offset == offset);
         return p;
     }
+    /* We may end up clearing live exceptions below, so make sure it's ours. */
+    assert(!PyErr_Occurred());
     do {
-        descr = _PyType_Lookup(type, p->name_strobj);
+        /* Use faster uncached lookup as we won't get any cache hits during type setup. */
+        descr = find_name_in_mro(type, p->name_strobj, &error);
         if (descr == NULL) {
+            if (error == -1) {
+                /* It is unlikely by not impossible that there has been an exception
+                   during lookup. Since this function originally expected no errors,
+                   we ignore them here in order to keep up the interface. */
```

Comment:
I don't think that "Since this function originally expected no errors" matters here. Why not reporting the exception to the caller and handle it?

Reply:
I'm working on a patch that cleans up much of the mess around `_PyType_Lookup()`. See https://bugs.python.org/issue31465